File: smartmall-frontend/src/services/shopService.js
import api from "./api";

export const shopService = {
	list: () => api.get("/shops"),
	get: (id) => api.get(`/shops/${id}`),
	create: (payload) => api.post("/shops", payload),
	update: (id, payload) => api.put(`/shops/${id}`, payload),
	remove: (id) => api.delete(`/shops/${id}`),
};

export default shopService;

File: smartmall-backend/src/main/java/com/smartmall/service/ShopService.java
package com.smartmall.service;

import com.smartmall.entity.Shop;

import java.util.List;

public interface ShopService {
	Shop create(Shop shop);

	List<Shop> getAll();

	Shop getById(Long id);

	Shop update(Long id, Shop shop);

	void delete(Long id);
}
